src/file_pipe_log/pipe.rs

```diff
@@ -254,6 +254,13 @@ impl<F: FileSystem> SinglePipe<F> {
         reader.read(handle)
     }
 
+    fn async_read(&self, block: &mut FileBlockHandle, ctx: &mut F::AsyncIoContext) {
+        let fd = self.get_fd(block.id.seq).unwrap();
+        let buf = vec![0_u8; block.len];
+
+        self.file_system.async_read(ctx, fd, buf, block).unwrap();
```

Comment:
Why not creating the vector inside `async_read`? So there's one less argument to pass into (and validate).